Doc/library/typing.rst

```diff
@@ -2815,6 +2815,56 @@ ABCs for working with IO
    represent the types of I/O streams such as returned by
    :func:`open`.
 
+The following protocols offer a simpler alternative for common use cases. They
+are especially useful for annotating function and method arguments and are
+decorated with :func:`@runtime_checkable <runtime_checkable>`.
+
+.. class:: Reader[T]
+
+   Protocol for reading from a file or other input stream.
+
+   .. method:: read(size=...)
+
+        Read data from the input stream and return it. If ``size`` is
+        specified, at most ``size`` items (bytes/characters) will be read.
+
+   .. method:: readline(size=...)
+
+        Read a line of data from the input stream and return it. If ``size`` is
+        specified, at most ``size`` items (bytes/characters) will be read.
+
+   .. method:: __iter__()
+
+        Read a line of data from the input stream and return it.
```

Comment:
This seems incorrect; it describes what calling `next` on the returned object will do, not what `__iter__` itself will do.